src/SimpleGraphs/generators/staticgraphs.jl

```diff
@@ -47,6 +47,61 @@ function CompleteBipartiteGraph(n1::T, n2::T) where {T <: Integer}
     return SimpleGraph(ne, fadjlist)
 end
 
+"""
+    CompleteMultipartiteGraph(partitions)
+
+Create an undirected [complete bipartite graph](https://en.wikipedia.org/wiki/Complete_bipartite_graph)
+with `sum(partitions)` vertices and `prod(partitions)` edges.
+"""
+function CompleteMultipartiteGraph(partitions::AbstractVector{T}) where {T <: Integer}
+    length(partitions) == 1 && return SimpleGraph{T}(partitions[1])
+    length(partitions) == 2 && return CompleteBipartiteGraph(partitions[1], partitions[2])
+    (length(partitions) == 0 || any(x -> x <= 0, partitions)) && return SimpleGraph(T)
+
+    Tw = widen(T)
+    nw = sum(partitions)
+    n = T(nw)  # checks if T is large enough for sum(partitions)
+
+    ne = 0
+    for p in partitions # type stability fails if we use sum and a generator here
+        ne += p*(Int(n)-p) # overflow if we don't convert to Int
+    end
+    ne = div(ne, 2)
+
+    fadjlist = Vector{Vector{T}}(undef, n)
+    cur = 1
+    for p in partitions
+        currange = cur:(cur+p-1) # all vertices in the current partition
+        lowerrange = 1:(cur-1)   # all vertices lower than the current partition
+        upperrange = (cur+p):n   # all vertices higher than the current partition
+        @inbounds @simd for u in currange
+            fadjlist[u] = vcat(Vector{T}(lowerrange), Vector{T}(upperrange))
+        end
+        cur += p
```

Comment:
I'd like to have this logic encoded into a test for `iscompletemultipartite` which asserts that the graph was constructed like this.